### PR TITLE
[query] Stage IndexReader

### PR DESCRIPTION
**Pull request by tpoterba**
look at that deleted code! Good riddance.

### CODE REVIEW

File: hail/src/main/scala/is/hail/io/index/StagedIndexReader.scala [hunk 1]
```diff
@@ -95,15 +95,11 @@ class StagedIndexReader(emb: EmitMethodBuilder[_], spec: AbstractIndexSpec) {
       endQuerySettable.asBaseStruct.loadField(cb, 0).get(cb).asInt64.value -
         startQuerySettable.asBaseStruct.loadField(cb, 0).get(cb).asInt64.value
     )
+    cb.assign(startLeaf, cb.memoize(startQuerySettable.asBaseStruct.loadField(cb, 1)))
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/io/index/StagedIndexReader.scala:98]: is the memoize call needed, aren't we assigning it to a variable anyways?

**Review by chrisvittal — DISMISSED** [review]
one nit, lgtm otherwise

File: hail/src/main/scala/is/hail/io/index/StagedIndexReader.scala (hunk 1 above)
Comment (tpoterba) [reply hail/src/main/scala/is/hail/io/index/StagedIndexReader.scala:98]: nope, fixed

**Review by chrisvittal — APPROVED** [review]
🎉

**Pull request merged**